### PR TITLE
Sign up validation

### PR DESCRIPTION
**Pull request by pmyip3**
# Description 
Add validation for teamName, orgName, email, password, and confirmed password in frontend before calling the createTeam API endpoint

# Type of change 
- [ ] Bug fix
- [X] New feature
- [ ] Refactoring
- [ ] Documentation 

# Problem
Currently, the program does not validate teamName, OrgName and the length of the passwords in the frontend 

# Solution description
Refractor code using formik with validations which can be found in the teamInfoSchema in client/src/components/auth/RegistrationForm.js 

# Tests 
All unit tests pass? N/A

Builds successfully? Yes

Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes)
image
image

Provide steps in detail for reviewers to reproduce the expected result of the code changes.

# Relevant document/ link (if any) 
Any related documents could help reviewers to review this PR 

# Risk
Low, only the register component and createTeam endpoint might be affected if bugs occurs

### CODE REVIEW

File: client/src/components/auth/RegistrationForm.js [hunk 1]
```diff
@@ -11,139 +11,153 @@ import { createTeam } from '../../actions/team';
...
+      .required('Team Name is required')
+      .min(3, 'Must be at least 3 characters'),
+    orgName: yup
+      .string()
+      .required('Organization Name is required')
+      .min(3, 'Must be at least 3 characters'),
+    email: yup
+      .string()
+      .email('Invalid Email')
+      .required('Email is required'),
```
Comment (vgoh1) [inline client/src/components/auth/RegistrationForm.js:40]: Can we improve this to have proper email domain validation? Not sure how complicated it'll get though

File: client/src/components/auth/RegistrationForm.js [hunk 2]
```diff
@@ -11,139 +11,153 @@ import { createTeam } from '../../actions/team';
...
+    password: yup
+      .string()
+      .required('Please Enter your password')
+      .matches(
+        /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/,
+        "Use 8 or more characters with a mix of letters, numbers & symbols"),
```
Comment (vgoh1) [inline client/src/components/auth/RegistrationForm.js:46]: For people who can't decode regex at a glance, what's the pattern here in plain words?

File: client/src/components/auth/RegistrationForm.js (hunk 1 above)
Comment (aksh0001) [reply client/src/components/auth/RegistrationForm.js:40]: If we can have password regex, we should be able to do it for email as well. The regex for email is widely documented online, e.g. on StackOverflow

**Review by aksh0001 — APPROVED** [review]

File: client/src/components/auth/RegistrationForm.js (hunk 1 above)
Comment (rpra0006) [reply client/src/components/auth/RegistrationForm.js:40]: Maybe have a check to see if '@' is at least present, as a valid email.

**Review by rpra0006 — APPROVED** [review]

File: client/src/components/auth/RegistrationForm.js (hunk 1 above)
Comment (pmyip3) [reply client/src/components/auth/RegistrationForm.js:40]: I think the built-in .email() function from formik already checking if '@' is present when validating the email.
In terms of checking the proper domain, I had a search but couldn't find any relevant regex. I don't think it's possible to do it cause a new email domain can be registered by anyone unless we want to be restricted the input email to a specific domain e.g. google.com.

File: client/src/components/auth/RegistrationForm.js (hunk 2 above)
Comment (vgoh1) [inline client/src/components/auth/RegistrationForm.js:46]: This message can be improved to be more specific about the password policy, that you need at least 1 digit and symbol on top of the characters.

**Review by vgoh1 — APPROVED** [review]

**Pull request merged**